bundles/org.openhab.core.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/service/ConfigurableServiceResource.java

```diff
@@ -310,10 +310,16 @@ private String getConfigDescriptionByFactoryPid(String factoryPid) {
     private String getServiceId(ServiceReference<?> serviceReference) {
         Object pid = serviceReference.getProperty(Constants.SERVICE_PID);
         if (pid != null) {
-            return (String) pid;
-        } else {
-            return (String) serviceReference.getProperty(ComponentConstants.COMPONENT_NAME);
+            if (pid instanceof String) {
+                return (String) pid;
+            } else {
+                String[] pidArray = (String[]) pid;
+                if (pidArray.length > 0) {
```

Comment:
Wouldn't it make sense, if we return the first one only and ignore all the others, to check if they are equal and if they differ log a warning that contains some information (component name, services pids, the used one).

Reply:
As mentioned in https://github.com/eclipse/smarthome/issues/6710#issuecomment-449592856, I do not at all understand why there should ever be different values. So for a start, this fixes the issue at hand.

Reply:
FTR, I also do not have a working IDE right now, so I cannot test changes thoroughly - only trying to fix the most critical issues asap, so that the distro gets usable again...

Question: While in fire-fighting mode, could we allow self-merging for fixing regressions to speed up things and address comments later on?